names/names.py

```diff
@@ -10,11 +11,35 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
-duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+# NESTED FOR LOOPS
+
+# duplicates = []                           O(1)
+# for name_1 in names_1:                    O(n)
+#     for name_2 in names_2:                O(n)
+#         if name_1 == name_2:              O(1)
+#             duplicates.append(name_1)     O(1)
+
+# TOTAL                                     O(2**n)
+# nested for runtime    6.037697076797485 seconds
+
+
+# BINARY SEARCH TREE
+
```

Comment:
This is great!, you're concentration for working towards an optimized time complexity will get you a long way!